.github/workflows/test-code-code_quality-kfp.yml

```diff
@@ -56,33 +56,26 @@ jobs:
                   sudo rm -rf /usr/share/dotnet /opt/ghc /usr/local/lib/android /usr/local/share/powershell /usr/share/swift /usr/lib/jvm /usr/local/.ghcup
                   sudo docker rmi $(docker image ls -aq) >/dev/null 2>&1 || true
                   df -h
+            - name: Import environment variables
+              run: |
+                  cat scripts/k8s-setup/requirements.env >> $GITHUB_ENV
+                  echo "K8S_SETUP_SCRIPTS=$PWD/scripts/k8s-setup" >> $GITHUB_ENV
+                  echo "REPOROOT=$PWD" >> $GITHUB_ENV
+                  echo "PATH=$PATH:/tmp" >> $GITHUB_ENV
             - name: Test V1 KFP workflow for transforms/code/code_quality
               timeout-minutes: 120
               run: |
+                  KFP_BLACK_LIST=$(./scripts/check-workflows.sh -show-kfp-black-list)
                   if [ -e "transforms/code/code_quality/Makefile" -a -e "transforms/code/code_quality/kfp_ray/Makefile" ]; then
-                      export REPOROOT=$PWD
-                      export K8S_SETUP_SCRIPTS=$PWD/scripts/k8s-setup
-                      source $K8S_SETUP_SCRIPTS/requirements.env
-                      export PATH=$PATH:/tmp/
-                      curl -Lo /tmp/kind https://kind.sigs.k8s.io/dl/v${KIND_VERSION}/kind-linux-amd64
-                      chmod 777 /tmp/kind
-                      curl -fsSL -o /tmp/get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
-                      chmod 700 /tmp/get_helm.sh
-                      HELM_INSTALL_DIR=/tmp/ /tmp/get_helm.sh -v v${HELM_VERSION} --no-sudo
-                      chmod 777 /tmp/helm
-                      curl -L https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl -o /tmp/kubectl
-                      chmod 777 /tmp/kubectl
-                      curl https://dl.min.io/client/mc/release/linux-amd64/mc --create-dirs -o /tmp/mc
-                      chmod +x /tmp/mc
-                      export DEPLOY_KUBEFLOW=1
-                      make -C $K8S_SETUP_SCRIPTS setup
-                      #make -C kfp/kfp_support_lib test
-                      make -C transforms/code/code_quality workflow-build
-                      source $K8S_SETUP_SCRIPTS/common.sh
-                      make -C transforms/code/code_quality workflow-test
-                      echo "Run transforms/code/code_quality completed"
-                   else
-                      echo "Skipping transforms/code/code_quality kfp test for lack of Makefile and/or kfp_ray/Makefile"
+                     transform=$(basename "transforms/code/code_quality")
+                     if echo ${KFP_BLACK_LIST} | grep -qv ${transform}; then
+                        $PWD/scripts/workflow_helper.sh install-tools
```

Comment:
can we have makefile targets that call these?  This would then allow the developer to run these tests locally, more easily and through mechanisms they are already familiar with (i.e. make).

Reply:
This would also force all the kfp tests to be run - they are not in the current PR .  See that only 3 checks were run.

Reply:
Would it be okay to address this in a separate issue? Personally, I'm not in favor of this approach because this install-tools function works specifically for Ubuntu, as used in the GitHub Action. The installation process differs for other operating systems, like macOS, with its various versions. I believe it would be simpler for each person to deploy or use their own tools on their machine, rather than creating a new script that we would need to maintain. However, I could be wrong on this...